### ISSUE TITLE
Spells dialogue gives odd answers

### ISSUE THREAD

**Issue by rdstern** [africanmathsinitiative/R-Instat#4724]
The spells dialogue has 2 problems - at least.

I tried with Dodoma from the library 

I got the spell lengths (unconditional) for January and then for February.  Then I got for January and February together.

1. The conditional spell lengths for January and February together gives zero every year!  At least that is obviously wrong!
2. The unconditional spell lengths look very sensible and reasonable.  And usually the reasult for Jan and Feb together is the same as the maximum for Jan and Feb.  But not always!  Why not?

**Comment by dannyparsons** [comment]
Could some of these detailed checks of results on climatic be tested by comparing to the old Instat? Maybe with help from @fran2or?

**Comment by rdstern** [comment]
Of course.  In this case I can get further with the problem as shown in the attached file:
[Samaru56_stacked.xlsx](https://github.com/africanmathsinitiative/R-Instat/files/2129883/Samaru56_stacked.xlsx)

Can of course alternatively use any climatic data set, e.g. Dodoma.
Then use Climatic > Prepare > Transform and then Spells.  This gives a column at the daily level of the spell lengths.  (This is already done in the Samaru file.)
Then use the Climatic > Check Data > Display Daily dialogue on the spell lengths column.  (It would be nice if we could control the number of decimals in that dialogue.)  Anyway set it to also display the maximum, because that is what the spells dialogue should also give us, if we set it to provide spells for complete months.

And looking at the results I think I have the answer!  R-Instat is giving the conditional spell lengths.  In the dialogue, when we tick to ask for conditional we get zeros, so that needs fixing too.  But I think what is said from the dialogue to be unconditional is (in fact) conditional.  So it is really the (usually more useful) unconditional that we don't have.

I hope that (or those) is now easy to fix.

**Comment by rdstern** [comment]
David and Beth have now worked on this and have found the calculations were incorrect.  The calculation "system" seems to be standing up well.  A special function was needed because the old calculations could not cope with the missing values correctly.

The new calculation will cope with the missing values and should now also produce the correct results for both unconditional and conditional spell lengths.  The name "conditional spell lengths" may be changing!

**Comment by shadrackkibet** [comment]
@rdstern I think Beth worked on this and it is now working better(I hope). Can you confirm that this is now working properly?. One problem i noticed is that  the conditional spells checkbox doesn't reset which we can fix.

**Comment by rdstern** [comment]
I am now in the section of the climatic guide where I need conditional spell lengths.  They now give an error, rather than the zeros we had earlier.  I hope this might be relatively simple to fix?
